packages/site/cms-site/src/sitePreview/SitePreviewApiHelper.ts

```diff
@@ -0,0 +1,57 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+type Scope = Record<string, unknown>;
+type GraphQLClient = {
+    setHeader(key: string, value: string): unknown;
+    request(document: string, variables?: unknown): Promise<{ isAllowedSitePreview: boolean }>;
+};
+export type SitePreviewParams = {
+    scope: Scope;
+    path: string;
+    settings: {
+        includeInvisible: boolean;
+    };
+};
+export type BlockPreviewParams = {
+    scope: Scope;
+    path: string;
+};
+
+export async function getValidatedSitePreviewParams(
+    req: NextApiRequest,
+    res: NextApiResponse,
+    graphQLClient: GraphQLClient,
+): Promise<SitePreviewParams> {
+    return {
+        scope: await getValidatedScope(req, res, graphQLClient),
+        path: req.query.path as string,
+        settings: JSON.parse(req.query.settings as string),
+    };
+}
+
+export async function getValidatedBlockPreviewParams(
```

Comment:
The separation of block/site/common preview is sometimes difficult to understand.

Here you put a BlockPreview related function into a file called SitePreviewApiHelper. Exaclty those parts cause the confusion. I would split it up into two files (site/block) or rename it into a gerenic preview.